test/builder/builder_test.cc

```diff
@@ -0,0 +1,122 @@
+//   and/or other materials provided with the distribution.
+//
+// * Neither the name of the copyright holder nor the names of its
+//   contributors may be used to endorse or promote products derived from
+//   this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#include "maliput_sparse/builder/builder.h"
+
+#include <memory>
+#include <string>
+
+#include <gtest/gtest.h>
+#include <maliput/api/junction.h>
+#include <maliput/api/lane.h>
+#include <maliput/api/road_geometry.h>
+#include <maliput/api/segment.h>
+
+namespace maliput_sparse {
+namespace builder {
+namespace {
+
+GTEST_TEST(Builder, Stub) {
+  const maliput::api::RoadGeometryId kRoadGeometryId("custom_rg_id");
+  constexpr double kLinearTolerance{1.};
+  constexpr double kAngularTolerance{2.};
+  constexpr double kScaleLength{2.};
+  const maliput::math::Vector3 kInertialToBackendFrameTranslation(4., 5., 6.);
+  const maliput::api::JunctionId kJunctionId("custom_j_id");
+  const maliput::api::SegmentId kSegmentAId("segment_a");
+  const maliput::api::LaneId kLaneAId("lane_a");
+  const maliput::api::LaneId kLaneBId("lane_b");
+  const maliput::api::SegmentId kSegmentBId("segment_b");
+  const maliput::api::LaneId kLaneCId("lane_c");
+
+  RoadGeometryBuilder dut;
+
+  // clang-format off
+  std::unique_ptr<maliput::api::RoadGeometry> rg =
+      dut.Id(kRoadGeometryId)
+          .LinearTolerance(kLinearTolerance)
+          .AngularTolerance(kAngularTolerance)
+          .ScaleLength(kScaleLength)
+          .InertialToBackendFrameTranslation(kInertialToBackendFrameTranslation)
+          .StartJunction()
+              .Id(kJunctionId)
+              .StartSegment()
+                      .Id(kSegmentAId)
+                  .StartLane()
+                      .Id(kLaneAId)
+                  .EndLane()
+                  .StartLane()
+                      .Id(kLaneBId)
+                  .EndLane()
+              .EndSegment()
+              .StartSegment()
+                  .Id(kSegmentBId)
+                  .StartLane()
+                      .Id(kLaneCId)
+                  .EndLane()
+              .EndSegment()
+          .EndJunction()
+          .Build();
+  // clang-format on
```

Comment:
What do you think about the following @francocipollone @stonier ?

The geometric information of each Lane has not been added yet but I have thought of 2 options:

1- Create all the LineString3d somwhere else and then attach those to each Lane as a property like ID.
2- Create a LaneGeometryBuilder and expose that extra Builder and let the LaneBuilder be agnostic of the geometric details.

Thoughts?

Reply:
> 1- Create all the LineString3d somwhere else and then attach those to each Lane as a property like ID.

We could be benefit of this approach for reusing line strings: --> Linestring could be the left boundaries for one a lane and the right boundary for another one.